apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/McsClaimTransformerV2.java

```diff
@@ -169,6 +170,62 @@ private Claim.ClaimStatus getStatus(RdaMcsClaim claimGroup) {
         : Claim.ClaimStatus.CANCELLED;
   }
 
+  /**
+   * Return the list of diagnosis codes wrapped in a {@link McsDiagnosisAdapterV2}.
+   *
+   * @param claimGroup claim group
+   * @return diagnosis adapter
+   */
+  private static Stream<McsDiagnosisAdapterV2> getDiagnosisCodes(RdaMcsClaim claimGroup) {
+    return ObjectUtils.defaultIfNull(claimGroup.getDiagCodes(), List.<RdaMcsDiagnosisCode>of())
+        .stream()
+        .sorted(Comparator.comparing(RdaMcsDiagnosisCode::getRdaPosition))
+        .map(McsDiagnosisAdapterV2::new);
+  }
+
+  /**
+   * Return the list of primary diagnosis codes wrapped in a {@link McsDiagnosisAdapterV2}.
+   *
+   * @param claimGroup claim group
+   * @return diagnosis adapter
+   */
+  private static Stream<McsDiagnosisAdapterV2> getPrimaryDiagnosisCodes(RdaMcsClaim claimGroup) {
+    return ObjectUtils.defaultIfNull(claimGroup.getDetails(), List.<RdaMcsDetail>of()).stream()
+        .sorted(Comparator.comparing(RdaMcsDetail::getIdrDtlNumber))
+        .map(McsDiagnosisAdapterV2::new);
+  }
+
+  /**
+   * Map the {@link McsDiagnosisAdapterV2} to a {@link Claim.DiagnosisComponent}.
+   *
+   * @param diagnosisAdapter adapter
+   * @return component
+   */
+  private static Claim.DiagnosisComponent getDiagnosisComponent(
+      McsDiagnosisAdapterV2 diagnosisAdapter) {
+
+    if (Strings.isNotBlank(diagnosisAdapter.getDiagnosisCode())) {
+      String system;
+
+      // Null check is necessary here because VALID_ICD_TYPES.contains(null) will throw a
+      // NullReferenceException
+      if (diagnosisAdapter.getIcdType() != null
+          && VALID_ICD_TYPES.contains(diagnosisAdapter.getIcdType())) {
+        system =
+            diagnosisAdapter.getIcdType().equals("0")
+                ? IcdCode.CODING_SYSTEM_ICD_10_CM
+                : IcdCode.CODING_SYSTEM_ICD_9;
```

Comment:
Is defaulting to ICD-9 the best option here?

Reply:
It's not defaulting - it checks against the list of valid types above which only allows `0` or `9`.

Reply:
Magnificent.

Reply:
Went ahead and refactored it anyway - I think a switch expression is more readable here (and avoids the pitfall that I mentioned in that comment).

Reply:
I lied - the null check is unfortunately still necessary (`switch (null)` throws an error since it calls `getHashCode` under the hood) but I think it's still more readable.